pandas/core/arrays/datetimelike.py

```diff
@@ -57,7 +58,8 @@ def _get_attributes_dict(self):
         return {k: getattr(self, k, None) for k in self._attributes}
 
     @property
-    def _scalar_type(self) -> Union[Type, Tuple[Type]]:
+    def _scalar_type(self) -> Union[Type[Period], Type[Timestamp],
```

Comment:
This is a restriction of the existing type annotation derived from the comment below.

Reply:
Why is the Type needed?  timestamp/period/timedelta are already classes...

Reply:
@jbrockmendel the difference between `Timestamp` and `Type[Timestamp]` would be that the former expects an instance of the `Timestamp` class whereas the latter expects the class itself

Reply:
Can you do `Type[Union[Period, Timestamp, Timedelta]]` hear instead? Maybe also make a Datetimelike alias in pandas._typing

Reply:
`Type` can only take one argument, so it has to be a `Union` of individual `Type`s.

But I can definitely add `DatetimeLike` in pandas._typing to replace it.

Reply:
Are you getting an error when trying to provide Type with a Union? It is a documented usage so curious what errors you might be getting:

https://docs.python.org/3/library/typing.html#typing.Type

Reply:
I got an error earlier complaining about only one argument, but maybe I inadvertently dropped the Union and just did `Type[Period Timestamp, Timedelta]`. With the Union it is not throwing an error.

Reply:
@gwrome you have this marked as resolved but still needs update I think

Reply:
I suppose pushing the update is key.